lib/active_job/jobs_relation.rb

```diff
@@ -51,14 +51,17 @@ def initialize(queue_adapter: ActiveJob::Base.queue_adapter, default_page_size:
   # * <tt>:queue_name</tt> - To only include the jobs in the provided queue.
   # * <tt>:worker_id</tt> - To only include the jobs processed by the provided worker.
   # * <tt>:recurring_task_id</tt> - To only include the jobs corresponding to runs of a recurring task.
-  def where(job_class_name: nil, queue_name: nil, worker_id: nil, recurring_task_id: nil)
+  # * <tt>:finished_at</tt> - (Range) To only include the jobs finished between de provided range
+  def where(job_class_name: nil, queue_name: nil, worker_id: nil, recurring_task_id: nil, finished_at: nil)
     # Remove nil arguments to avoid overriding parameters when concatenating +where+ clauses
     arguments = { job_class_name: job_class_name,
       queue_name: queue_name,
       worker_id: worker_id,
-      recurring_task_id: recurring_task_id
+      recurring_task_id: recurring_task_id,
     }.compact.collect { |key, value| [ key, value.to_s ] }.to_h
 
+    arguments.merge!(finished_at: finished_at) if finished_at.present? && !finished_at.to_s.match?(/^..$/)
```

Comment:
not sure about this, for example:
```ruby
dummy(dev)> hash = { finished_at: nil..nil }.compact
=> {:finished_at=>nil..}
dummy(dev)> hash.collect { |key, value| [ key, value.to_s ] }.to_h
=> {:finished_at=>".."}

```
will override finished_at to an empty range

Reply:
I think that's not an empty range, but an "infinite" range, a range without start and end.

Reply:
besides, I did not want not stringity the range, check out this behaviour:
```
dummy(dev)> ActiveJob.jobs.finished.where(finished_at: (..Time.now).to_s).first
  SolidQueue::Job Load (0.2ms)  SELECT "solid_queue_jobs".* FROM "solid_queue_jobs" WHERE "solid_queue_jobs"."finished_at" IS NOT NULL AND "solid_queue_jobs"."finished_at" = ? ORDER BY "solid_queue_jobs"."finished_at" DESC LIMIT ? OFFSET ?  [["finished_at", "2024-11-17 19:53:22"], ["LIMIT", 1000], ["OFFSET", 0]]
=> nil
dummy(dev)> ActiveJob.jobs.finished.where(finished_at: ..Time.now).first
  SolidQueue::Job Load (0.5ms)  SELECT "solid_queue_jobs".* FROM "solid_queue_jobs" WHERE "solid_queue_jobs"."finished_at" IS NOT NULL AND "solid_queue_jobs"."finished_at" <= ? ORDER BY "solid_queue_jobs"."finished_at" DESC LIMIT ? OFFSET ?  [["finished_at", "2024-11-17 19:53:28.197854"], ["LIMIT", 1000], ["OFFSET", 0]]
=>

```

Reply:
Why would you convert the range into a string? That wouldn't work in the query.

Reply:
https://github.com/rails/mission_control-jobs/blob/a8073e71f9db1eacdce35f46199d9d5a70923dd1/lib/active_job/jobs_relation.rb#L61

this line will do the `.to_s`

Reply:
and sorry yes you are right, we could just do: `arguments.merge!(finished_at: finished_at)` without the condition I wrote

Reply:
Ah, sorry! Yes, we don't want to do that to the range. My comment was more addressed at this part: 
```ruby
if finished_at.present? && !finished_at.to_s.match?(/^..$/)
```
I meant we don't need that and we could just initialize `arguments[:finished_at]` with the value. 

In any case, I think we can get rid of the `to_s` call on the values 🤔 It was added [here](https://github.com/rails/mission_control-jobs/commit/e6f0ad185c52853b855d8f4710b839b68604bd86) because we didn't have the `compact` before, but I don't think we need it. You can try to remove it and see if any tests fail.

Reply:
great!! thanks :)